openpype/hosts/resolve/api/lib.py

```diff
@@ -266,28 +272,41 @@ def create_timeline_item(media_pool_item: object,
     clip_name = _clip_property("File Name")
     timeline = timeline or get_current_timeline()
 
+    # timing variables
+    fps = project.GetSetting("timelineFrameRate")
+    duration = source_end - source_start
+    timecode_in = frames_to_timecode(timeline_in, fps)
+    timecode_out = frames_to_timecode(timeline_in + duration, fps)
+
     # if timeline was used then switch it to current timeline
     with maintain_current_timeline(timeline):
         # Add input mediaPoolItem to clip data
-        clip_data = {"mediaPoolItem": media_pool_item}
-
-        # add source time range if input was given
-        if source_start is not None:
-            clip_data.update({"startFrame": source_start})
-        if source_end is not None:
-            clip_data.update({"endFrame": source_end})
+        clip_data = {
+            "mediaPoolItem": media_pool_item,
+            "startFrame": source_start,
+            "endFrame": source_end,
+            "recordFrame": timeline_in,
+        }
 
+        print("clip_data", "_" * 50)
+        print(media_pool_item.GetName())
         print(clip_data)
+
         # add to timeline
         media_pool.AppendToTimeline([clip_data])
 
         output_timeline_item = get_timeline_item(
             media_pool_item, timeline)
 
-    assert output_timeline_item, AssertionError(
-        "Track Item with name `{}` doesn't exist on the timeline: `{}`".format(
-            clip_name, timeline.GetName()
-        ))
+    assert output_timeline_item, AssertionError((
+        "Clip name '{}' was't created on the timeline: '{}' \n\n"
+        "Please check if the clip is in the media pool or if the timeline \n"
+        "is having activated correct track name, or if it is not already \n"
+        "having any clip add place on the timeline in: '{}' out: '{}'. \n\n"
+        "Clip data: {}"
+    ).format(
+        clip_name, timeline.GetName(), timecode_in, timecode_out, clip_data
+    ))
```

Comment:
```suggestion
    assert output_timeline_item, AssertionError((
        "Clip name '{}' wasn't created on the timeline: '{}' \n\n"
        "Please check if the clip is in the media pool or if the timeline \n"
        "has correct track name activated, or if it does not already \n"
        "have any clip add place on the timeline in: '{}' out: '{}'. \n\n"
        "Clip data: {}"
    ).format(
        clip_name, timeline.GetName(), timecode_in, timecode_out, clip_data
    ))
```

Even with these grammar tweaks suggested I'm not sure what the last OR statement means. If it doesn't have any clip on the timeline yet, what should I do? And what did I do wrong if I hit that particular case?

Reply:
Yeah, I have almost broken my brain too.. here is fix a810dfe78a97eb1467d3745ee786f2743f6faac2